.github/workflows/ci.yml

```diff
@@ -81,3 +81,26 @@ jobs:
           # Token is not required for public repos, but see:
           # https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954
           token: ${{ secrets.CODECOV_TOKEN }}
+
+  test-publish:
+    runs-on: ubuntu-latest
+    needs: run-tests
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.4.1
+
+      - name: Build a binary wheel and a source tarball
+        run: poetry build
+
+      - name: Test package is publishable with PyPI test server
+        uses: pypa/gh-action-pypi-publish@v1.10.3
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          skip-existing: true
```

Comment:
Glad they've added this; the inability to skip-existing was the only reason we had the test in the deploy stage in the the first place (which didn't really make sense). Although of course this only tests once per potential release but it's better than nothing.